### PR TITLE
more detailed test output (see #909)

### PR DESCRIPTION
**Pull request by rbri**
i know this is a bit of a hack but i like to focus on fixing this cases instead of making super cool code here

### CODE REVIEW

**Comment by gbrail** [comment]
Thanks -- this is helpful. 

How do you feel about having it always print out the list of tests that didn't pass in all cases, versus only printing if a system property or something is set? We'll have to look at those messages for quite a long time as I suspect that some of them will take a while to fix.

**Comment by rbri** [comment]
@tonygermano, @tuchida, @p-bakker what do you think?

File: testsrc/org/mozilla/javascript/tests/Test262SuiteTest.java [hunk 1]
```diff
@@ -509,4 +530,117 @@ static Test262Case fromSource(File testFile) throws IOException {
...
+
+        public void passes(int optLevel, boolean useStrict) {
+            if (useStrict) {
+                strictOptLevel.remove(optLevel);
+            } else {
+                nonStrictOptLevel.remove(optLevel);
+            }
+        }
+
+        public String getMessage(int[] optLevels) {
```
Comment (tuchida) [inline testsrc/org/mozilla/javascript/tests/Test262SuiteTest.java:559]: Isn't `Optional<String> getMessage` better?
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

**Comment by p-bakker** [comment]
Very useful indeed.

Do wonder if a different format would be better (and doable), one where the results aren't encoded in a string like `fails in strict mode but passes in non-strict mode for all opt level(s) -1, 0, 9`, but where I could take to output and parse it (for example a csv format and pasting it into Excel) and then analyse it.

For example to find all tests that fail on all optLevels in either strict or non-strict mode or find all tests that always fail in non-interpreted mode

**Comment by gbrail** [comment]
Thinking more about this...

I'm afraid that if we add lots of output to a "successful" build then we'll end up masking other problems because people will just ignore everything.

I would prefer if this instead printed out one message like this:

Warning:
27 tests failed in strict mode but passed in non-strict mode
99 tests failed for some optimization levels and passed for others
See buildGradle/reports/inconsistent-tests.txt for detail

**Comment by rbri** [comment]
#930 is the future

**Pull request closed without merging**